New set of refactor

rodrigosiqueira opened the pull request:
This PR has part of Alan's patches and other improvements to the deploy part.

kwy95 commented on tests/deploy_test.sh:
```diff
@@ -420,12 +463,12 @@ function test_kernel_install_local()
   }
 
-  output=$(kernel_install "1" "test" "TEST_MODE" "2")
+  output=$(run_kernel_install "1" "test" "TEST_MODE" "2")
```
Quotes

kwy95 commented on tests/deploy_test.sh:
```diff
@@ -544,29 +571,26 @@ function test_kernel_uninstall()
-  expected_cmd[7]="$kernel_uninstall_cmd"
+  output=$(run_kernel_uninstall 3 1 "$single_kernel" 'TEST_MODE')
+  cmd="bash $remote_path/remote_deploy.sh --uninstall_kernel 1 remote $single_kernel TEST_MODE"
+  run_kernel_uninstall_cmd="ssh -p 3333 juca@127.0.0.1 sudo \"$cmd\""
```
```suggestion
  run_kernel_uninstall_cmd="ssh -p 3333 juca@127.0.0.1 sudo '$cmd'"
```

kwy95 reviewed (changes requested):
Seems good, I requested a few changes, mostly grammatical and some quotes in lines you are already changing anyway.

rodrigosiqueira commented on tests/deploy_test.sh:
```diff
@@ -168,66 +165,117 @@ function test_kernel_install()
+  # The following commands represets those steps
+  local cmd_preset_remote="$rsync_cmd $preset_path $remote:/root/kw_deploy $rsync_flags"
+  local cmd_image_remote="$rsync_cmd $kernel_image_path $remote:$kernel_image_remote_path $rsync_flags"
+  local cmd_deploy_image="$ssh_cmd $remote sudo \"$deploy_cmd\""
```
Here I'm testing the command output; in other words, I'm comparing strings. However, we can probably update the deploy function and next update this test. We can create an issue for tracking that.

rodrigosiqueira commented on src/deploy.sh:
```diff
@@ -253,6 +268,75 @@ function parse_deploy_options()
+function prepare_host_deploy_dir()
+{
+  if [[ -z "$KW_CACHE_DIR" ]]; then
+    complain "\$KW_CACHE_DIR isn't set. The kw directory at home may not exist"
```
Don't worry about that right now. I have another PR that will remove a lot of these things.

rodrigosiqueira commented on src/deploy.sh:
```diff
@@ -253,6 +268,75 @@ function parse_deploy_options()
+    return 22
+  fi
+
+  # We should expect the setup.sh script create the directory $HOME/kw.
+  # However, does not hurt check for it and create in any case
```
I'm going to skip this change because I have another PR refactoring this function.

rodrigosiqueira commented on src/deploy.sh:
```diff
@@ -253,6 +268,75 @@ function parse_deploy_options()
+  distro_info=$(which_distro "$remote" "$port" "$user")
+  distro=$(detect_distro '/' "$distro_info")
+
+  if [[ $distro =~ "none" ]]; then
+    complain "Unfortunately, there's no support for the target distro"
```
My next PR will address this problem.

rodrigosiqueira commented:
Hi @kwy95 , in this new version, I addressed most of your comments. I did not apply any of your suggestions to the prepare_remote and prepare_host function because I have another series that will specifically focus on those parts; I'll send it after we land this change.

rodrigosiqueira commented:
Merged,
Thanks @kwy95
